core-rust/state-manager/src/store/db.rs

```diff
@@ -338,3 +344,114 @@ impl RecoverableVertexStore for StateManagerDatabase {
         }
     }
 }
+
+pub enum WriteableStateManagerDatabaseAccountChangeIndex<'a> {
+    InMemory(WriteableInMemoryAccountChangeIndex<'a>),
+    RocksDB(WriteableRocksDBAccountChangeIndex<'a>),
+}
+
+impl WriteableStoreIndexExtension for WriteableStateManagerDatabaseAccountChangeIndex<'_> {
+    fn disable(&mut self) {
```

Comment:
I'm rather surprised that the index has dynamic enable/disable functionality - you can't dynamically turn on the index else it's incomplete, which would lead to confusion - unless we carefully track which state versions from/to it's been enabled for, and expose that very carefully via the API.

I imagine this is quite complicated to do for now, and we should just ensure either it's enabled or disabled.

I would have assumed this comes from the StateManager settings at start-up time.

Reply:
Also how are enable and disable even called?